tartiflette/engine.py

```diff
@@ -96,79 +96,65 @@ def __init__(
-                """
-            )
+        self._schema_name = schema_name
+        self._error_coercer = error_coercer
+        self._custom_default_resolver = custom_default_resolver
+        self._modules = modules
+        self._sdl = sdl
 
     async def cook(
         self,
-        sdl: Union[str, List[str]],
+        sdl: Union[str, List[str]] = None,
         error_coercer: Callable[[Exception], dict] = None,
         custom_default_resolver: Optional[Callable] = None,
         modules: Optional[Union[str, List[str]]] = None,
-        schema_name: str = "default",
+        schema_name: str = None,
     ):
         """
         Cook the tartiflette, basicly prepare the engine by binding it to given modules using the schema_name as a key.
         You wont be able to execute a request if the engine wasn't cooked.
 
-        Arguments:
-            sdl {Union[str, List[str]]} -- The SDL to work with.
-
         Keyword Arguments:
+            sdl {Union[str, List[str]]} -- The SDL to work with.
             schema_name {str} -- The name of the SDL (default: {"default"})
             error_coercer {Callable[[Exception, dict], dict]} -- An optional callable in charge of transforming a couple Exception/error into an error dict (default: {default_error_coercer})
             custom_default_resolver {Optional[Callable]} -- An optional callable that will replace the tartiflette default_resolver (Will be called like a resolver for each UNDECORATED field) (default: {None})
             modules {Optional[Union[str, List[str]]]} -- An optional list of string containing the name of the modules you want the engine to import, usually this modules contains your Resolvers, Directives, Scalar or Subscription code (default: {None})
         """
 
         if not modules:
-            modules = []
+            modules = self._modules or []
 
         if isinstance(modules, str):
             modules = [modules]
 
+        schema_name = schema_name or self._schema_name or "default"
+
+        custom_default_resolver = (
+            custom_default_resolver or self._custom_default_resolver
+        )
+        if custom_default_resolver and not iscoroutinefunction(
+            custom_default_resolver
+        ):
+            raise Exception(
+                f"Given custom_default_resolver "
+                f" {custom_default_resolver} "
+                f"is not a coroutine function"
+            )
+
         self._error_coercer = error_coercer_factory(
-            error_coercer or default_error_coercer
+            error_coercer or self._error_coercer or default_error_coercer
         )
+
         self._modules, modules_sdl = await _import_modules(
             modules, schema_name
         )
-        SchemaRegistry.register_sdl(schema_name, sdl, modules_sdl)
+
+        SchemaRegistry.register_sdl(schema_name, sdl or self._sdl, modules_sdl)
```

Comment:
SDL is required but it might be `None` if we didn't pass it to `Engine` nor `.cook()`.

You may want to add a check similar to:

```python
if self._sdl is None:
    assert sdl, "'sdl' is required because none was given when instanciating the engine"
```

Reply:
I'll add a sanity check here, but not with assert :)